### ISSUE TITLE
MPS: scaled_dot_product_attention returns wrong output shape when value dim != query/key dim

### ISSUE THREAD

**Issue by han-so1omon** [pytorch/pytorch#176767]
## Description
On the MPS backend (Apple Silicon), torch.nn.functional.scaled_dot_product_attention appears to return an output whose last dimension matches the query/key head dimension instead of the value head dimension when the two differ.

According to the PyTorch documentation, the output shape of scaled_dot_product_attention should be:

```py
(..., L, Ev)
```

where Ev is the last dimension of the value tensor.

However, when running the operation on the MPS backend, the output instead has the last dimension equal to the query/key head dimension.

This behavior does not occur on CPU, where the output shape correctly follows the value dimension.

This suggests a backend-specific issue in the MPS implementation of SDPA.

⸻

## Minimal Repro

```py
import torch
import torch.nn.functional as F

print("torch version:", torch.__version__)
print("MPS available:", torch.backends.mps.is_available())

q = torch.randn(1, 1, 16, 32)
k = torch.randn(1, 1, 16, 32)
v = torch.randn(1, 1, 16, 24)

# CPU reference
y_cpu = F.scaled_dot_product_attention(
    q, k, v,
    dropout_p=0.0,
    is_causal=False,
)

print("CPU output shape:", y_cpu.shape)

if torch.backends.mps.is_available():
    q = q.to("mps")
    k = k.to("mps")
    v = v.to("mps")

    y_mps = F.scaled_dot_product_attention(
        q, k, v,
        dropout_p=0.0,
        is_causal=False,
    )

    print("MPS output shape:", y_mps.shape)
```

## Expected Behavior

The output last dimension should match the value tensor.

```
CPU output shape: torch.Size([1, 1, 16, 24])
MPS output shape: torch.Size([1, 1, 16, 24])
```

## Actual Behavior

On MPS, the output shape matches the query/key head dimension instead of the value dimension.

Example output:
```
CPU output shape: torch.Size([1, 1, 16, 24])
MPS output shape: torch.Size([1, 1, 16, 32])
```

## Notes
	•	The issue appears only when value.shape[-1] != query.shape[-1].
	•	When Ev == E, the behavior matches expectations.
	•	The same code produces correct results on CPU.

### Versions

PyTorch version: 2.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.2 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.6.3.2)
CMake version: version 3.29.5
Libc version: N/A

Python version: 3.11.15 | packaged by conda-forge | (main, Mar  5 2026, 16:59:26) [Clang 19.1.7 ] (64-bit runtime)
Python platform: macOS-26.2-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M3 Pro

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk
